src/compartment.c

```diff
@@ -130,6 +130,9 @@ comp_from_elf(char* filename, struct ConfigEntryPoint* entry_points, size_t entr
         }
         else if (new_comp->elf_type == ET_EXEC)
         {
+            // TODO maybe just remove this if if we don't want to support
+            // static binaries anymore
+            assert(false);
```

Comment:
`assert` can be optimised out so `err("...")` or similar might be better?

Reply:
I'll throw in these replies commits which tackle adding `err`s. It's gonna be a bunch of them. I can squash them into one, perhaps, if you prefer. I'll say when I think I'll be done.

https://github.com/capablevms/CHERI-ELF-comp/pull/13/commits/1ea365a9984d958e2d1a9fed758530fc567b59e2
https://github.com/capablevms/CHERI-ELF-comp/pull/13/commits/569506d85d7900debb57337fdd785962cd4d02d2
https://github.com/capablevms/CHERI-ELF-comp/pull/13/commits/c53f6260f2904b774ca0f315b2d66ad46d015e02

Reply:
This should be the last commit changing all exit points with `err`s - https://github.com/capablevms/CHERI-ELF-comp/pull/13/commits/0baac864ab873c385b5458bfdd920e8ede355369.